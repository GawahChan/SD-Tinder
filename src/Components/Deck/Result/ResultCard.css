.result-card {
    border-radius: 10px;
    box-shadow: 1px 1px 8px #333;
    margin: 10px;

    background-size: cover;
    background-position: top;
    overflow: hidden;

    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    
    min-width: 250px;
    min-height: 250px;
}

.result-card-title {
    color: white;
    background: linear-gradient(rgba(0,0,0,0), rgba(0,0,0,1));
    padding: 10px;
}


.result-card-title h2 {
    margin: 0;
    font-weight: 400;
}