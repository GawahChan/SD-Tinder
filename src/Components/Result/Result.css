.result {
    text-align: center;
    margin: 30px 0 30px 0;
}

.result h1 {
    font-weight: 400;
    font-size: 2.5rem;
}

.result h2 {
    font-weight: 400;
    font-size: 1.5rem;
}

.result h3 {
    font-weight: 400;
    font-size: 1rem;
    margin: 0;
}

.result-profiles {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
}