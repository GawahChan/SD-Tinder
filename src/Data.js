const Data = [
    {id: 0, name: 'Mila Kunis', age: '30', location: 'LA', bio: 'Actress' }
];

export default Data;