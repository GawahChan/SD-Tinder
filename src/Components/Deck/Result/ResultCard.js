import React from 'react';
import './ResultCard.css';

function ResultCard(props) {
    return (
        <div className='result-card' style={{ 'backgroundImage': `url(${require(`../../../Common/Profile/${props.img}.jpg`)})` }}>
            <div className='result-card-title'>
                <h2>{props.name}</h2>
            </div>
        </div>
    );
}

export default ResultCard;