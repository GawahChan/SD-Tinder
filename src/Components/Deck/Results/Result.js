import React from "react";
import ResultCard from "./ResultCard";
import { Title, Subtitle } from "../../../Common/Styled/Typography.styles";
import { ResultDisplay, Container } from "./style";

function Result(props) {
  return (
    <Container>
      <Title>{props.title}</Title>
      <ResultDisplay>
        {props.stuntDouble.length > 0 ? (
          props.stuntDouble.map(stuntDouble => {
            return (
              <ResultCard
                key={stuntDouble.id}
                img={stuntDouble.img}
                name={stuntDouble.name}
              />
            );
          })
        ) : (
          <Subtitle>{props.subTitle}</Subtitle>
        )}
      </ResultDisplay>
    </Container>
  );
}

export default Result;
