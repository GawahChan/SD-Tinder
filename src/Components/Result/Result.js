import React from 'react';
import ResultCard from './ResultCard.js';
import './Result.css';

function Result(props) {
    return (
        <div className='result'>
            <h1>There's no one new around you</h1>
            <div className='result-superliked'>
                <h2>People you Superliked!</h2>
                <div className='result-profiles superliked'>
                    {
                        props.superliked.length > 0 ?
                            props.superliked.map(stuntDouble => {
                                return (
                                    <ResultCard
                                        key={stuntDouble.id}
                                        img={stuntDouble.img}
                                        name={stuntDouble.name}
                                    />
                                )
                            }) :
                            <h3>You didn't superlike any profiles</h3>
                    }
                </div>
            </div>
            <div className='result-liked'>
                <h2>People you liked... let's hope they like you too!</h2>
                <div className='result-profiles liked'>
                    {
                        props.liked.length > 0 ?
                            props.liked.map(stuntDouble => {
                                return (
                                    <ResultCard
                                        key={stuntDouble.id}
                                        img={stuntDouble.img}
                                        name={stuntDouble.name}
                                    />
                                )
                            }) :
                            <h3>You didn't like any profiles</h3>
                    }
                </div>
            </div>
            <div className='result-disliked'>
                <h2>People you disliked</h2>
                <div className='result-profiles disliked'>
                    {
                        props.disliked.length > 0 ?
                            props.disliked.map(stuntDouble => {
                                return (
                                    <ResultCard
                                        key={stuntDouble.id}
                                        img={stuntDouble.img}
                                        name={stuntDouble.name}
                                    />
                                )
                            }) :
                            <h3>Cast a wide net they said...</h3>
                    }
                </div>
            </div >
        </div>
    );
}

export default Result;