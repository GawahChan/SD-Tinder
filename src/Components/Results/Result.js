import React from "react";
import PropTypes from "prop-types";
import { ThemeProvider } from "styled-components";
import { theme } from "../../Theme";
import ResultCard from "./ResultCard";
import { Title, Subtitle } from "../../Common/Styled/Typography.styles";
import { ResultDisplay, ResultContainer } from "./style";

function Result({ id, title, stuntDouble, subTitle }) {
  return (
    <ThemeProvider theme={theme[id]}>
      <ResultContainer>
        <Title>{title}</Title>
        <ResultDisplay>
          {stuntDouble.length > 0 ? (
            stuntDouble.map(stuntDouble => {
              return (
                <ResultCard
                  key={stuntDouble.id}
                  img={stuntDouble.img}
                  name={stuntDouble.name}
                />
              );
            })
          ) : (
            <Subtitle>{subTitle}</Subtitle>
          )}
        </ResultDisplay>
      </ResultContainer>
    </ThemeProvider>
  );
}

Result.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  stuntDouble: PropTypes.array,
  subTitle: PropTypes.string
};

export default Result;
