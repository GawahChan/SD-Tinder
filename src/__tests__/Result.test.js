import React from 'react';
import { shallow } from 'enzyme';
import Result from '../Components/Result/Result';
import Data from '../Data';

describe('Result component', () => {
    let props = { liked: [Data[0]], superliked: [Data[1]], disliked: [Data[2]]}
    let result = shallow(<Result {...props}/>);

    it('renders correctly', () => {
        expect(result).toMatchSnapshot();
    });

    it('renders <ResultCard/> once for each prop', () => {
        expect(result.find('ResultCard').length).toEqual(3);
        expect(result.find('.result-profiles.superliked').children('ResultCard').length).toEqual(1);
        expect(result.find('.result-profiles.liked').children('ResultCard').length).toEqual(1);
        expect(result.find('.result-profiles.disliked').children('ResultCard').length).toEqual(1);
    });

    describe('Result component where all props have no data', () => {
        let props = { liked: [], superliked: [], disliked: []}
        let result = shallow(<Result {...props}/>);

        it('does not render <ResultCard/> for .result-propiles.superliked', () => {
            expect(result.find('.result-profiles.superliked').children('ResultCard').length).toEqual(0);
            expect(result.find('.result-profiles.superliked').children('h3').length).toEqual(1);
        });
        it('does not render <ResultCard/> for .result-propiles.liked', () => {
            expect(result.find('.result-profiles.liked').children('ResultCard').length).toEqual(0);
            expect(result.find('.result-profiles.liked').children('h3').length).toEqual(1);
        })
        it('does not render <ResultCard/> for .result-propiles.disliked', () => {
            expect(result.find('.result-profiles.disliked').children('ResultCard').length).toEqual(0);
            expect(result.find('.result-profiles.disliked').children('h3').length).toEqual(1);
        })
    });








})