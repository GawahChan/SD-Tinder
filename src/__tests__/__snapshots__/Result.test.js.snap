// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result component renders correctly 1`] = `
<div
  className="result"
>
  <h1>
    There's no one new around you
  </h1>
  <div
    className="result-superliked"
  >
    <h2>
      People you Superliked!
    </h2>
    <div
      className="result-profiles superliked"
    >
      <ResultCard
        img="Mila-Kunis"
        key="1"
        name="Mila Kunis"
      />
    </div>
  </div>
  <div
    className="result-liked"
  >
    <h2>
      People you liked... let's hope they like you too!
    </h2>
    <div
      className="result-profiles liked"
    >
      <ResultCard
        img="Rosie-Mac"
        key="0"
        name="Rosie Mac"
      />
    </div>
  </div>
  <div
    className="result-disliked"
  >
    <h2>
      People you disliked
    </h2>
    <div
      className="result-profiles disliked"
    >
      <ResultCard
        img="Alicia-Vela-Bailey"
        key="2"
        name="Alicia Vela-Bailey"
      />
    </div>
  </div>
</div>
`;
